web/controllers/password_reset_controller.ex

```diff
@@ -11,14 +11,14 @@ defmodule CodeCorps.PasswordResetController do
   and return email. 422 if pwd do not match or auth token does not exist
   """
   def reset_password(conn, %{"token" => token, "password" => password, "password_confirmation" => password_confirmation}) do
-    user = conn.assigns.current_user
-    with %AuthToken{value: auth_token} <- Repo.get_by(CodeCorps.AuthToken, %{ value: token, user_id: user.id }),
+    with %AuthToken{value: auth_token, user_id: user_id} <- Repo.get_by(CodeCorps.AuthToken, %{ value: token }),
       {:ok, _} <- Phoenix.Token.verify(CodeCorps.Endpoint, "user", auth_token, max_age: 1209600) do
-        with %Changeset{valid?: true} <- User.reset_password_changeset(user, 
-                                                                       %{password: password, password_confirmation: password_confirmation}) do
+        user = Repo.get(User, user_id)
+        with %Changeset{valid?: true} = changeset <- User.reset_password_changeset(user, 
+                                                                       %{password: password, password_confirmation: password_confirmation}),
+             {:ok, user} <- Repo.update(changeset) do
           conn
-          |> put_status(:created)
-          |> render("show.json", email: user.email)
+          |> TokenController.create(%{"username" => user.email, "password" => password})
```

Comment:
@begedin So one thing I noticed this weekend was when calling the TokenController from this controller, it still used the PasswordResetView (I assumed it would use the token controller).  So in this case, the token controller will return the token and user id.  However, the frontend would need the `email` also since the password reset route only knows about the `password/passwordConfirmation` in order to login right away.